src/mongo/db/storage/wiredtiger/wiredtiger_record_store.h

```diff
@@ -277,9 +287,9 @@ class WiredTigerRecordStore : public RecordStore {
     const bool _isEphemeral;
     // True if the namespace of this record store starts with "local.oplog.", and false otherwise.
     const bool _isOplog;
-    const int64_t _cappedMaxSize;
+    int64_t _cappedMaxSize;
     const int64_t _cappedMaxSizeSlack;  // when to start applying backpressure
-    const int64_t _cappedMaxDocs;
```

Comment:
You are right，OplogStones parameter should be changed when resize oplog, like belows.

```
void WiredTigerRecordStore::OplogStones::adjust() {
    stdx::lock_guard<stdx::mutex> lk(_mutex);
    unsigned long long maxSize = _rs->cappedMaxSize();

    const unsigned long long kMinStonesToKeep = 10ULL;
    const unsigned long long kMaxStonesToKeep = 100ULL;

    unsigned long long numStones = maxSize / BSONObjMaxInternalSize;
    _numStonesToKeep = std::min(kMaxStonesToKeep, std::max(kMinStonesToKeep, numStones));
    _minBytesPerStone = maxSize / _numStonesToKeep;
    invariant(_minBytesPerStone > 0);
}   bool WiredTigerRecordStore::setCappedSize(long long cappedSize) {
    _cappedMaxSize = cappedSize;
    if (_oplogStones) {
        _oplogStones->adjust();
    }
    return true;
}

bool WiredTigerRecordStore::setCappedSize(long long cappedSize) {
    _cappedMaxSize = cappedSize;
    if (_oplogStones) {
        _oplogStones->adjust();
    }
    return true;
}

```